Engine/main/main.cpp

```diff
@@ -313,6 +314,10 @@ static int main_process_cmdline(ConfigTree &cfg, int argc, char *argv[])
             play.takeover_from[49] = 0;
             ee += 2;
         }
+        else if (ags_stricmp(arg, "--clear-cache-on-room-change") == 0)
+        {
+            usetup.clear_cache_on_room_change = true;
```

Comment:
It's best to instead write a value into the `cfg` object, to let it be merged with the config and applied uniformly.

`INIwriteint(cfg, "misc", "clear_cache_on_room_change", 1);`

Reply:
Thanks, I will do that then. I have left the PR as draft until I can adjust this.

Reply:
There's a mix of usetup, cfg, and INIwrite usages in main.cpp, I don't understand what are the differences between each of these ways of doing things. Perhaps a comment on top of the file could clarify when to use what.